src/Microsoft.ML.StandardTrainers/Standard/LogisticRegression/MulticlassLogisticRegression.cs

```diff
@@ -993,35 +949,282 @@ DataViewRow ICanGetSummaryAsIRow.GetSummaryIRowOrNull(RoleMappedSchema schema)
 
         DataViewRow ICanGetSummaryAsIRow.GetStatsIRowOrNull(RoleMappedSchema schema)
         {
-            if (_stats == null)
+            if (Statistics == null)
                 return null;
 
             VBuffer<ReadOnlyMemory<char>> names = default;
-            var meta = _stats.MakeStatisticsMetadata(null, schema, in names);
+            var meta = Statistics.MakeStatisticsMetadata(null, schema, in names);
             return AnnotationUtils.AnnotationsAsRow(meta);
         }
     }
 
     /// <summary>
-    /// A component to train a logistic regression model.
+    /// Linear model of multiclass classifiers. It outputs raw scores of all its linear models, and no probablistic output is provided.
     /// </summary>
-    public partial class LogisticRegressionBinaryClassificationTrainer
+    public sealed class LinearMulticlassModelParameters : LinearMulticlassModelParametersBase
     {
-        [TlcModule.EntryPoint(Name = "Trainers.LogisticRegressionClassifier",
-            Desc = Summary,
-            UserName = LogisticRegressionMulticlassClassificationTrainer.UserNameValue,
-            ShortName = LogisticRegressionMulticlassClassificationTrainer.ShortName)]
-        internal static CommonOutputs.MulticlassClassificationOutput TrainMulticlass(IHostEnvironment env, LogisticRegressionMulticlassClassificationTrainer.Options input)
+        internal const string LoaderSignature = "MulticlassLinear";
+        internal const string RegistrationName = "MulticlassLinearPredictor";
+
+        private static VersionInfo VersionInfo =>
+            new VersionInfo(
+                modelSignature: "MCLINEAR",
+                // verWrittenCur: 0x00010001, // Initial
+                // verWrittenCur: 0x00010002, // Added class names
+                verWrittenCur: 0x00010003, // Added model stats
+                verReadableCur: 0x00010001,
```

Comment:
As far as I know this is a new model type. Why is this not the initial version?

Reply:
Thanks for pointing this out. Just fixed it.

---
In reply to: [266553016](https://github.com/dotnet/machinelearning/pull/2976#discussion_r266553016) [](ancestors = 266553016)